digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -553,3 +561,79 @@ fn filter_out_stale_creators(creators: &mut Vec<asset_creators::Model>) {
         }
     }
 }
+
+pub fn get_edition_data_from_json<T: DeserializeOwned>(data: Value) -> Result<T, DbErr> {
+    serde_json::from_value(data).map_err(|e| DbErr::Custom(e.to_string()))
+}
+
+pub async fn get_nft_editions(
+    conn: &impl ConnectionTrait,
+    mint_address: Pubkey,
+    limit: Option<u32>,
+    page: Option<u32>,
+) -> Result<NftEditions, DbErr> {
+    let master_edition_pubkey = MasterEdition::find_pda(&mint_address).0;
+
+    let master_edition =
+        asset_v1_account_attachments::Entity::find_by_id(master_edition_pubkey.to_bytes().to_vec())
+            .one(conn)
+            .await?
+            .ok_or(DbErr::RecordNotFound(
+                "Master Edition not found".to_string(),
+            ))?;
+
+    let limit = limit.unwrap_or(10);
+
+    let master_edition_data: MasterEdition = master_edition
+        .data
+        .clone()
+        .ok_or(DbErr::RecordNotFound(
+            "Master Edition data not found".to_string(),
+        ))
+        .map(get_edition_data_from_json)??;
+
+    let nft_editions = asset_v1_account_attachments::Entity::find()
+        .filter(
+            asset_v1_account_attachments::Column::AttachmentType
+                .eq(V1AccountAttachments::Edition)
+                .and(asset_v1_account_attachments::Column::Data.is_not_null())
+                .and(Expr::cust(&format!(
+                    "data->>'parent' = '{}'",
+                    master_edition_pubkey
+                ))),
+        )
+        .order_by_asc(asset_v1_account_attachments::Column::SlotUpdated)
+        .all(conn)
+        .await?;
+
+    let nft_editions = nft_editions
+        .iter()
+        .map(|e| -> Result<NftEdition, DbErr> {
+            let data: Edition = e
+                .data
+                .clone()
+                .ok_or(DbErr::RecordNotFound("Edition data not found".to_string()))
+                .map(get_edition_data_from_json)??;
```

Comment:
Its normal to chain the map to the query result and to implement a TryInto for the conversion so code looks like

```
    let nft_editions = asset_v1_account_attachments::Entity::find()
        .filter(
            asset_v1_account_attachments::Column::AttachmentType
                .eq(V1AccountAttachments::Edition)
                .and(asset_v1_account_attachments::Column::Data.is_not_null())
                .and(Expr::cust(&format!(
                    "data->>'parent' = '{}'",
                    master_edition_pubkey
                ))),
        )
        .order_by_asc(asset_v1_account_attachments::Column::SlotUpdated)
        .all(conn)
        .await?
        .into_iter()
        .map(TryInto::try_into)
        .collect::<Result<..>>()?;

```